### DISCUSSION TITLE
Does it work with Net 9?

### DISCUSSION THREAD

**Discussion by mdelgadov** [reactiveui/refit#1935] [Q&A] [answered]
I had it running very well in.Net 8 Maui and ported to the new Net 9 and it swallows the response...

I upgraded to Refit 8 but still have got the same problem.

It goes to the interface as follows:

global::System.Threading.Tasks.Task<global::System.Net.Http.HttpResponseMessage> global::ZoEazy.Model.Interfaces.IAuthApi.AvailableEmailAsync(global::ZoEazy.Model.Contracts.EmailValidator @email) 
        {
            var ______arguments = new object[] { @email };
            var ______func = requestBuilder.BuildRestResultFuncForMethod("AvailableEmailAsync", new global::System.Type[] { typeof(global::ZoEazy.Model.Contracts.EmailValidator) } );
            return (global::System.Threading.Tasks.Task<global::System.Net.Http.HttpResponseMessage>)______func(this.Client, ______arguments);
        }

it goes to the return, gets the request (using debug, ofc) and never returns to the end bracket...

Of course, .Net 9 and Refit 8 are three weeks old, but have to ask:

Is .Net 9 supported?

Thanks in advance

**Comment by dev-fJ-del** [comment]
I've upgraded my Maui proj to .net 9, refit to v8.0.0
Seems to have an issue deserializing `Ardalis.SmartEnum` whereas in .net 8 it works perfectly fine.

> {Refit.ApiException: An error occured deserializing the response.
>  ---> System.TypeLoadException: Could not set up parent class, due to: Recursive type definition detected

**Reply by mdelgadov** [reply]
In Net9, I removed Refit and it all worked well, it swallows the response to not be seen again.. and freezes the app.... kinda black hole...

**Reply by ChrisPulman** [reply]
I am currently working abroad at the moment so I don't have as much availability as normal, but I will try to get net 9.0 compatibility added as soon as possible.

**Comment by Syed-RI** [comment, marked as answer]
Seems to be working fine in .NET9. I think it just needs new TFM and no dotnet 9 specific work is required here (just a guess)

**Comment by glennawatson** [comment]
Yes, Refit supports .NET 9 (and .NET 10). Current source multi-targets net8.0, net9.0, and net10.0, and the test suite runs on .NET 9.

If a call "swallows the response" and never returns, that is almost always a deadlock from blocking on async code, not a Refit/net9 incompatibility. Common causes:

- Calling `.Result` or `.Wait()` on the returned Task somewhere up the chain (UI thread deadlock on MAUI).
- Awaiting without `ConfigureAwait` while a parent frame blocks.

Make the whole call chain async/await end to end and never block on the Task.

For the separate `Ardalis.SmartEnum` "Recursive type definition detected" report: that is a converter/serializer issue, not Refit. It reproduces with the same converter outside Refit on the newer runtime; configure the SmartEnum JSON converter in your `RefitSettings.ContentSerializer`.

If you still hit an issue on current Refit, please open a new issue with a minimal repro.
